alphadia/transferlearning/metrics.py

```diff
@@ -79,22 +67,29 @@ def calculate_test_metric(self, test_input: dict, epoch: int):
         epoch : int
             The epoch at which the test metric is calculated.
 
+        dataset : str
+            The name of the dataset. e.g. "train", "validation", "test".
+
+        property_name : str
+            The name of the property. e.g. "charge", "rt"
+
         Returns
         -------
-        pd.DataFrame
-            A pandas dataframe containing the test metric at the given epoch.
+        list[dict]
+            A list of dictionaries containing the test metric at the given epoch in the long format.
+            e.g. [{"dataset": "train", "epoch": 1, "property": "charge", "metric_name": "accuracy", "value": 0.9}]
```

Comment:
I think we can already have them as pandas dataframes right from the begininng. No need to have the list of dicts pattern.